### ISSUE TITLE
Agent to Agent direct calls fail

### ISSUE THREAD

**Issue by jsandersj** [twilio-professional-services/flex-dialpad-addon-plugin#21]
Good morning!

I have been attempting to deploy the plugin for a few days without luck. The issue I'm seeing is likely similar to closed issue #12 . The plugin successfully deploys, the serverless functions successfully deploy, and I do see the option in the drop down menu from dialpad to select an agent:

![image](https://user-images.githubusercontent.com/7594508/115750720-1933a380-a35e-11eb-91b8-9e713433a6e8.png)

I select an agent from drop down, and it attempts to place the outbound call. however, I don't see anything at the destination agent side. I don't see anything show up on live logging in the service, but if I view the Chrome console, 

I see this:

twilio-flex.min.js:1604 POST https://event-bridge.twilio.com/v1/wschannels 400 (myURL-dev.twil.io/internal-call/agent-outbound-join?taskSid=WTmyTaskis not a valid CallUrl)

slightly redacted. In actual output, full URL is there and full task WT is present. I have validated the path in question is present, and it shows the following code:

```
exports.handler = (context, event, callback) => {
    let twiml = new Twilio.twiml.VoiceResponse();
  
        twiml.dial().conference({
          statusCallback: 'call-outbound-join',
          statusCallbackEvent: 'join end',
          endConferenceOnExit: true,
        }, event.taskSid);
    
      callback(null, twiml);
  };
```

Based on previous notes on issue #12  it sounded like the issue is usually with environment variables. re: the serverless deployed functions, I do show the three required environment variables on the functions service for TWILIO_WORKFLOW_SID, TWILIO_WORKSPACE_SID, and TWILIO_NUMBER.

Note that if i check I do see a task does appear be created. We see assignment status reserved, and the reservation shows as below:

![image](https://user-images.githubusercontent.com/7594508/115752749-0621d300-a360-11eb-97d6-620068ae0f80.png)

note that the worker jsanders here is the originator of the call attempt.

Editing to add one more piece of detail. The workflow referenced in the .env file is shown below if there are any issues with this:

![image](https://user-images.githubusercontent.com/7594508/115753170-82b4b180-a360-11eb-9559-ad036bd42e6a.png)

Any suggestions/guidance would be greatly appreciated! thank you!

**Comment by jsandersj** [comment]
UPDATE: Working this issue, believe we have found two issues. One, the Workflow, matching tasks showed targetworker != null. This is a syntax error. it should be targetWorker != null. 

Second, when assigning the plugin .env file has this environment variable: REACT_APP_SERVICE_BASE_URL. Per deployment notes, we should get the domain from deploying and insert here. However, the value for domain returned for the deployment does not include a leading "https://". ie, it showed somedomain.io but for REACT_APP_SERVICE_BASE_URL, it does need to include a leading https://. So where the return from deployment is somedomain.io, the environment variable should be https://somedomain.io. 

That's an issue on my side, but may be worth adding a note for clarity on instructions so for people like me not as familiar with the process to save them some pain :)

Once we corrected these, it does seem to work. One question/clarification: We cannot call an agent unless they are in an Available state, correct? We can select any agent in a state besides Offline when selecting a destination, but it does not seem to be delivered unless the target is in available state.
